internal/pubsub/samples/pub/main.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// pub is a sample application that publishes messages from stdin to an
+// existing topic.
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"gocloud.dev/internal/pubsub"
+)
+
+func main() {
+	log.SetFlags(0)
+	log.SetPrefix("pub: ")
+	envFlag := flag.String("env", "", "environment [gcp|rabbit]")
+	flag.Parse()
+	if *envFlag == "" {
+		log.Fatal("missing -env flag\n")
+	}
+	if flag.NArg() != 1 {
+		fmt.Fprintf(os.Stderr, "usage: pub -env=<env> [flags] <topic-name>\n")
```

Comment:
Is there a bug yet about `pubsub` supporting URLs? That would simplify this, it would be more like:

`pub gcspubsub://my-topic`

Reply:
Filed https://github.com/google/go-cloud/issues/1025 for the more general question of whether to support URLs throughout Go Cloud.

Reply:
There doesn't seem to be an issue for that specific to pubsub.